addons/mail/static/src/discuss/voice_message/common/voice_recorder.js

```diff
@@ -82,6 +82,10 @@ export class VoiceRecorder extends Component {
                 this.microphone = await browser.navigator.mediaDevices.getUserMedia({
                     audio: this.userSettings.audioConstraints,
                 });
+                if (status(this) === "destroyed") {
```

Comment:
Why not registering a callback on `onWillDestroy` to *always* release the microphone?

Reply:
@tsm-odoo `onWillDestroy` could be called before the `getUserMedia` promise is resolved so at that time microphone could be `undefined`, then the promise is resolved (after `onWillDestroy`) and the microphone is not "closed".

Reply:
@tsm-odoo Yes it would be equivalent, it seemed that cleanup did a lot of additional unnecessary things for that case